src/v/raft/heartbeat_manager.h

```diff
@@ -136,6 +136,15 @@ class heartbeat_manager {
     bool is_stopped() const { return _bghbeats.is_closed(); }
 
 private:
+    struct heartbeat_requests {
+        /// Requests to dispatch.  Can include request to self.
+        std::vector<heartbeat_manager::node_heartbeat> requests;
```

Comment:
why not a flat_hash_set, growth factor?

Reply:
this vector is usually small, and we do not do a lot of lookups into it